### PR TITLE
[tflite] allow gpu delegate to run quant model in tflite demo

### PR DESCRIPTION
**Pull request by freedomtan**
GPU delegate can handle quantized model since 40088b87ee0

### CODE REVIEW

File: tensorflow/lite/java/demo/app/src/main/java/com/example/android/tflitecamerademo/Camera2BasicFragment.java [hunk 1]
```diff
@@ -368,12 +368,7 @@ private void updateActiveModel() {
       classifier.setNumThreads(numThreads);
       if (device.equals(cpu)) {
       } else if (device.equals(gpu)) {
-        if (model.equals(mobilenetV1Quant)) {
```
Comment (jdduke) [inline tensorflow/lite/java/demo/app/src/main/java/com/example/android/tflitecamerademo/Camera2BasicFragment.java:371]: Do we also have to call setQuantizedModelsAllowed on the GpuDelegate?

**Review by jdduke — CHANGES_REQUESTED** [review]

File: tensorflow/lite/java/demo/app/src/main/java/com/example/android/tflitecamerademo/Camera2BasicFragment.java (hunk 1 above)
Comment (freedomtan) [reply tensorflow/lite/java/demo/app/src/main/java/com/example/android/tflitecamerademo/Camera2BasicFragment.java:371]: Yes, my bad. I thought it was enabled by default in related Java code. Updated.

**Review by srjoglekar246 — COMMENTED** [review]
Looks good to me. @freedomtan Can you post some before/after numbers from your testing device in the PR?

**Comment by freedomtan** [comment]
@srjoglekar246 Before the PR, as expected, no numbers were reported when trying to `mobilenet v1 quant` on GPU. If allowing it to be run without calling `setQuantizedModelsAllowed()`, TFLite interpreter number was reported (because failing to apply GPU delegate). After `setQuantizedModelsAllowed()`, as expected, I got numbers close ones I got when running the float model. The app doesn't report stable average numbers. Roughly, I got

| device | CPU 1xthread | GPU delegate|
|----------:|-------------------:|-------------------:|
| Pixel 4 | 28 ms             | 13 ms|
| Dimensity 1000+ dev board | 14 ms             | 7 ms|
| Dimensity 820 dev board | 15 ms             | 15 ms|

Note that numbers on dev boards are for reference only :-) The CPU number of Pixel 4 should be much better. But that's what I got from the app. I actually got much better number from `benchmark_model`.

**Comment by srjoglekar246** [comment]
@freedomtan Awesome, thanks for confirming!

**Review by srjoglekar246 — APPROVED** [review]

File: tensorflow/lite/java/demo/app/src/main/java/com/example/android/tflitecamerademo/ImageClassifier.java [hunk 2]
```diff
@@ -171,8 +171,15 @@ private void recreateInterpreter() {
   }
 
   public void useGpu() {
+    useGpu(false);
+  }
+
+  public void useGpu(boolean allowQuantizedModels) {
```
Comment (srjoglekar246) [inline tensorflow/lite/java/demo/app/src/main/java/com/example/android/tflitecamerademo/ImageClassifier.java:177]: Do we need this boolean & the method above? We can just always set the param to true?

**Review by srjoglekar246 — DISMISSED** [review]

File: tensorflow/lite/java/demo/app/src/main/java/com/example/android/tflitecamerademo/ImageClassifier.java (hunk 2 above)
Comment (jdduke) [reply tensorflow/lite/java/demo/app/src/main/java/com/example/android/tflitecamerademo/ImageClassifier.java:177]: Agreed, let's just always set to true. We'll probably deprecate/remove that flag in a future update and just always allow quantized execution.

Comment (freedomtan) [reply tensorflow/lite/java/demo/app/src/main/java/com/example/android/tflitecamerademo/ImageClassifier.java]: updated. thanks.

**Review by srjoglekar246 — APPROVED** [review]

**Pull request merged**